### PR TITLE
Deprecate RxJava

### PR DESCRIPTION
**Pull request by nomisRev**
Since Arrow Fx Coroutines is going to be the new way forward over `IO` or polymorphic effect typeclasses the Arrow Fx RxJava module is deprecated in favor of using Arrow Fx Coroutines + KotlinX Coroutines RxJava.

```kotlin
import kotlinx.coroutines.rx2.mono
import kotlinx.coroutines.rx2.asCoroutineDispatcher
import io.reactive.Scheduler

val single: Single<Pair<String, String>> = single {
  parMapN(Schedulers.computation().asCoroutineDispatcher(),
    { Thread.currentThread().name },
    { Thread.currentThread().name },
    ::Pair
  )
}
```

### CODE REVIEW

**Review by aballano — APPROVED** [review]

**Review by franciscodr — APPROVED** [review]
Thanks @nomisRev!

File: arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt [hunk 1]
```diff
@@ -245,6 +275,7 @@ data class FlowableK<out A>(val flowable: Flowable<out A>) : FlowableKOf<A> {
     fun <A> cancelableF(fa: ((Either<Throwable, A>) -> Unit) -> FlowableKOf<CancelToken<ForFlowableK>>, mode: BackpressureStrategy = BackpressureStrategy.BUFFER): FlowableK<A> =
```
Comment (danimontoya) [inline arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt:275]: The deprecation message should be change for `DeprecateRxJava` ?

File: arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt [hunk 2]
```diff
@@ -228,6 +257,7 @@ data class FlowableK<out A>(val flowable: Flowable<out A>) : FlowableKOf<A> {
     fun <A> cancelable(fa: ((Either<Throwable, A>) -> Unit) -> CancelToken<ForFlowableK>, mode: BackpressureStrategy = BackpressureStrategy.BUFFER): FlowableK<A> =
```
Comment (danimontoya) [inline arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/FlowableK.kt:257]: The deprecation message should be change for `DeprecateRxJava` ?

**Review by i-walker — APPROVED** [review]

**Review by danimontoya — APPROVED** [review]
Some functions were already deprecated because of renaming.. maybe those deprecation messages need to be changed to `DeprecateRxJava` ?

**Pull request merged**